### ISSUE TITLE
Own-line `ty:ignore` and unknown rule, fix never converges

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#4032]
### Summary

`main.py`:

```py
seen = True
# ty: ignore[not-a-rule]
value = 1
```

Applying the fix to suppress the warning on `not-a-rule` keeps adding new suppressions, without suppressing the violation

[Playground](https://play.ty.dev/1bc4788e-a934-4f2c-a88d-f46eb5e85029)

**Comment by charliermarsh** [comment]
Ack, thanks!
